tests/test_emulator_retry.py

```diff
@@ -0,0 +1,125 @@
+
+    def test_root(self):
+        response = self.client.get("/")
+        self.assertEqual(response.data, b"OK")
+
+    def test_retry_test_crud(self):
+        self.assertIn("storage.buckets.list", emulator.db.supported_methods)
+        response = self.client.post(
+            "/retry_test",
+            data=json.dumps({"instructions": {"storage.buckets.list": ["return-429"]}}),
+        )
+        self.assertEqual(response.status_code, 200, msg=response.data)
+        self.assertTrue(
+            response.headers.get("content-type").startswith("application/json")
+        )
+        create_rest = json.loads(response.data)
+        self.assertIn("id", create_rest)
+
+        response = self.client.get("/retry_test/" + create_rest.get("id"))
+        self.assertEqual(response.status_code, 200)
+        self.assertTrue(
+            response.headers.get("content-type").startswith("application/json")
+        )
+        get_rest = json.loads(response.data)
+        self.assertEqual(get_rest, create_rest)
+
+        response = self.client.get("/retry_tests")
+        self.assertEqual(response.status_code, 200)
+        self.assertTrue(
+            response.headers.get("content-type").startswith("application/json")
+        )
+        list_rest = json.loads(response.data)
+        ids = [test.get("id") for test in list_rest.get("retry_test", [])]
+        self.assertEqual(ids, [create_rest.get("id")], msg=response.data)
+
+        response = self.client.delete("/retry_test/" + create_rest.get("id"))
+        self.assertEqual(response.status_code, 200)
+        # Once deleted, getting the test should fail.
+        response = self.client.get("/retry_test/" + create_rest.get("id"))
+        self.assertEqual(response.status_code, 404)
+
+    def test_retry_test_create_invalid(self):
+        response = self.client.post("/retry_test", data=json.dumps({}))
+        self.assertEqual(response.status_code, 400)
+
+    def test_retry_test_get_notfound(self):
+        response = self.client.get("/retry_test/invalid-id")
+        self.assertEqual(response.status_code, 404)
+
+    def test_retry_test_return_error(self):
+        response = self.client.post(
+            "/retry_test",
+            data=json.dumps({"instructions": {"storage.buckets.list": ["return-429"]}}),
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertTrue(
+            response.headers.get("content-type").startswith("application/json")
+        )
+        create_rest = json.loads(response.data)
+        self.assertIn("id", create_rest)
+
+        list_response = self.client.get(
+            "/storage/v1/b",
+            query_string={"project": "test-project-unused"},
+            headers={"x-retry-test-id": create_rest.get("id")},
+        )
+        self.assertEqual(list_response.status_code, 429, msg=list_response.data)
+
+    def test_retry_test_return_error_after(self):
```

Comment:
Missed this last time. IIUC, I don't think `retry_return_error_after_bytes` is fully supported in the Retry Test API yet. The assertion here will still pass with 200.

Reply:
You mean I should revert the changes to `common.py` to support this feature?  I can do that, but I did implement the stuff correctly I think.

Reply:
I'm not sure, but looking at [`common.handle_retry_test_instruction`](https://github.com/googleapis/storage-testbench/blob/c3f07fc5ac95fcadcd436bd4bf0f09079c49aa5d/testbench/common.py#L348), is the Retry Test API able to handle `retry_return_error_after_bytes` yet? It seems like it only handles `retry_return_error_code` for now.

Also, there's an [open issue](https://github.com/googleapis/google-cloud-cpp/issues/6615).

Reply:
Argh... Removed.  I got confused because I have a pending change that fixes that bug:

https://github.com/googleapis/storage-testbench/commit/a1d6ceb659e75e78796db051e4804d963080fa41#diff-58348f821aff24198cd8ec2b8e3f9ae89229a984392217de030c266fbf56ce89R416

Reply:
Wow that's awesome!!! Thanks so much for helping add that functionality. And sorry for confusing you 😂